docs(api): use named arguments in code samples

jwwojak opened the pull request:
<!--
Thanks for taking the time to open a pull request! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview

This PR is for RTC-32, Use Named Arguments in Code Samples. We've reviewed and inventoried all the code samples in the API docs. Most of the API methods in the tutorial use positional arguments. To help improve clarity for users, we're going to add the argument names to all the code samples.

The inventory is attached to the JIRA issue and linked here: [Named arguments](https://docs.google.com/document/d/1FE0jlKSh4OWpWMdrvU02Jyup4N8iiOd7xtpqwREyBgQ/edit#)

Opening as a draft PR. Will finish and push all revisions here before making this a real/regular PR and asking for reviews. 

This is a low-priority, time-permitting project. This PR may stay as a draft for a while.

# Test Plan

1. Format original code samples from the Opentrons API docs using the [Black Python code formatter](https://black.readthedocs.io/en/latest/).
2. Upload reformatted `.py` files in the Opentrons app for an automated check. If files pass the check, push to the draft PR.
3. After reviewing and pushing all the code, ask a willing engineer to review the reformatted code.
4. Make any recommended changes.
5. Change PR from a draft to regular PR and do the usual process. 

# Changelog

All the code samples in the API docs will change from using positional arguments to named arguments. Affects sample code examples in our online docs only.

Example
- Original code snippet: `tiprack = protocol.load_labware('opentrons_96_tiprack_300ul', 1)`
- Revised code snippet: `tiprack = protocol.load_labware(load_name= 'opentrons_96_tiprack_300ul', location= 1)`

# Review requests

Does this changed formatting look weird, not in line with standard practices, or incorrect?

**Sandbox URL**
http://sandbox.docs.opentrons.com/named-arguments-for-api-docs/v2/

# Risk assessment

Moderate. Even though this PR changes documentation code examples and snippets only, if revisions aren't accurate, code won't work for users attempting to learn how to use the Opentrons API. Code that doesn't work will contribute to user frustration and distrust with our documentation.

jwwojak committed:
API Tutorial Code Sample Revisions

jwwojak committed:
Merge branch 'edge' into named-arguments-for-api-docs

jwwojak committed:
Revisions to new_labware.rst

Adds named arguments to the code samples in the Labware chapter. Revisions also formatted to avoid horizontal scrolling.

jwwojak committed:
Merge branch 'edge' into named-arguments-for-api-docs

jwwojak committed:
Labware

Revised code samples for the labware chapter. Also, went back to 2 other pages (index and tutorial) to make sure spacing between things like variables and = are uniform within and among all the examples.

jwwojak committed:
Revisions to new_modules.rts

Continuing the named argument code work.

Also, added a few small text change suggestions. For example:

- In "Available Modules, revise intro sentences in 1st paragraph.
- In "Using a Temperature Module", put ending text into a note callout box.

jwwojak committed:
Update new_modules.rst

Continuing to add argument names to methods in the examples. For new_modules.rst

jwwojak committed:
new_pipette revisions - ongoing

Afternoon save. Revisions in progress.

jwwojak committed:
Finished chapter

new_pipette.rst code samples revised.

Building block chapter next.

jwwojak committed:
Merge branch 'edge' into named-arguments-for-api-docs

jwwojak committed:
Revisions to new atomic commands

Adding named arguments to code samples in new_atomic_commands.rst.

jwwojak committed:
More revisions to new atomic commands

Finishing changes to code samples for Building Block Commands page (new_atomic_commands.rst).

jwwojak committed:
Revisions to Complex Commands

Added names to method arguments for Complex Commands (new_complex_commands.rst).

jwwojak committed:
Merge branch 'edge' into named-arguments-for-api-docs

jwwojak committed:
Revisions to Position Within the OT-2

Adds names to method arguments in code samples. This is for the chapter, Position Within the OT-2 (robot_position.rst).

jwwojak committed:
Revisions to Examples chapter

Revising code samples to show named arguments in methods. This starts the update to the Examples chapter (new_examples.rst).

jwwojak committed:
Merge branch 'edge' into named-arguments-for-api-docs

jwwojak committed:
Revisions to Examples chapter

Finishing code sample revisions for this chapter (new_examples.rst).  Show named arguments in code samples.

jwwojak committed:
Merge branch 'edge' into named-arguments-for-api-docs

jwwojak committed:
Review push

A quick push to save review changes, just in case.

jwwojak committed:
Final chnanges

After a review, I had made some argument names "destination" when they should have been "dest".

And other revisions to the complext commands doc.

jwwojak commented:
Changing to ready for review PR. Sandbox URL in description above and here: 

- http://sandbox.docs.opentrons.com/named-arguments-for-api-docs/v2/

cc: @ecormany

sfoster1 commented on api/docs/v2/index.rst:
```diff
@@ -56,29 +56,31 @@ For example, if we wanted to have the OT-2 transfer liquid from well A1 to well
-        plate = protocol.load_labware('corning_96_wellplate_360ul_flat', location='1')
-        tiprack = protocol.load_labware('opentrons_96_tiprack_300ul', location='2')
+        plate = protocol.load_labware(
+            load_name = "corning_96_wellplate_360ul_flat", location = "1")
```
this may be too annoying to handle, but the typical python coding style (and what we enforce with tooling in our own code) is that named arguments do not have a space between argument-name, the equals sign, and the value: `load_name="corning_96_wellplate_360ul_flat"`. This doesn't affect functionality at all, but most other python tutorials and content will have this style.

ecormany replied:
yep, that's [what `black` does](https://black.vercel.app/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AFqALxdAD2IimZxl1N_WmOMs4-wxpT_MUF7H3K2siNmTii_aEUfIBKUIwcIRHj6eHJTTO3-ldDy1V07n6kM4-gRpl7Dul_tV9g2YHVydJ063gQ6fiVxYXgSd3-iyohcMntJdEafNwT5UCDafPxeF_XZIZgyVmhEYBdswmV45W_1JQ7UhCqdfMvLdSQDuzHFCZHLK3AWdoUM92TW9lqzxNjFE2Op1J6M9zIIAeyp0Nc5X5g2RwoiZF27jNnB8sm2WFYAAIawe5Y8YpgEAAHYAesCAAB4caICscRn-wIAAAAABFla) 

i thought whether there was some way to quickly find all of these with a regex and…if there is it's beyond me. probably quickest to search for ` = ` and blast through the document manually replacing with `=` inside args. there's about 200 of 'em.

jwwojak replied:
I deliberately put spaces between = for clarity. The code works even if this isn't a best practice. The idea was to err on the side of making it visually easy to read and understand, particularly for users who are just getting started with API stuff. For users with more experience, well, they might find it annoying, but know that the extra spaces aren't needed.

jwwojak replied:
New push to remove extra spaces around `=` symbol.

jwwojak committed:
Fix code sample in Complex Commands

The "Distribute" section in Complex Commands includes  a code snippet that isn't correct, this:

dest=plate.rows()['A']

And should be dest=plate.rows_by_name()['A']

jwwojak committed:
Remove spaces around =

Added spaces around the = symbol for visual clarity. Removing extra space to conform to typical code style.

Example:
Remove this: plate = protocol.load_labware
Replace with: plate=protocol.load_labware

ecormany commented:
To clarify Seth's earlier comment, `black` formatting puts spaces around an equals sign that does variable assignment but doesn't put spaces around equals signs for named arguments. Is this stupid? Maybe! Arbitrary? Certainly! Should we follow it? I think so…and that means the latest commit has actually removed too many spaces.

jwwojak committed:
Afternoon revisions

Following Black format, strictly. Restore space around variables and =. Named arguments don't get spaces around =. Saving work so far. Incomplete.

Files revised today:
- index.rst
- new_labware.rst
- tutorial.rst

The Black formatting style sometimes gives us a dedented or outdented closing bracket on its own line, but that's how it works.

jwwojak committed:
Update new_modules.rst

Continuing revisions. Aligning samples with Black standards. This is new_modules.rst.
